### DISCUSSION TITLE
Breakpoints

### DISCUSSION THREAD

**Discussion by mimarz** [digdir/designsystemet#5291] [Best practices]
Collect experiences and recommendations for an article people can reference as suggested values for breakpoints.

Since media queries don't support variables, we will only make a table with suggestions for people to copy values.

Notes:
- https://tailwindcss.com/docs/responsive-design
- https://mantine.dev/styles/responsive/
- https://react-spectrum.adobe.com/react-spectrum/Provider.html#breakpoints
- https://aksel.nav.no/grunnleggende/styling/brekkpunkter
- https://v2.chakra-ui.com/docs/styled-system/theme#breakpoints
- https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility/#media-queries-7
- https://keithjgrant.com/posts/2023/05/px-vs-em-in-media-queries/

**Comment by mimarz** [comment]
Maybe these breakpoints?

|Size   	| em   	| px   	|
|---	|---	|---	|
|xs   	| 40em  	|640px   	|
|s   	| 48em  	|768px   	|
|m   	| 64em  	| 1024px  	|
|l  	| 80em  	| 1280px  	|
|xl 	|  96em 	| 1536px  	|

**Comment by Camulos** [comment]
Seem like there is a lot of users that use resolutions close to the minimums mentioned in WCAG (320px width). There is also something strange about the stats and most likely there is some pollution in the source-data used in Statcounter. I think the smallest size need to work from minimum 320 px, but we do not need to test for usability at smaller screens than 320.
https://gs.statcounter.com/screen-resolution-stats/all/norway 

Would be interesting to see comparable results from some public service websites.

At BR we do not track viewport at the moment as this is not gathered on the server-side statistics that we get (in Kibana). However, I have asked for operation system and browser stats.

**Comment by ingvilddev** [comment]
> Seem like there is a lot of users that use resolutions close to the minimums mentioned in WCAG (320px width). There is also something strange about the stats and most likely there is some pollution in the source-data used in Statcounter. I think the smallest size need to work from minimum 320 px, but we do not need to test for usability at smaller screens than 320. https://gs.statcounter.com/screen-resolution-stats/all/norway
> 
> Would be interesting to see comparable results from some public service websites.
> 
> At BR we do not track viewport at the moment as this is not gathered on the server-side statistics that we get (in Kibana). However, I have asked for operation system and browser stats.

I've tried to ask the designers in NRK TV if they make Figma sketches based on actual data in terms of if they know what screen sizes the users are usually on. I got as a reply per 22nd of Jan 2026 that they just use the Figma presets ⬇️ as a starting point.

<img width="289" height="384" alt="Image" src="https://github.com/user-attachments/assets/776e5f4e-fb66-48bb-b0e7-74ac46cee529" />



We unfortunately have not put it into system (although I would have guessed it would be nice for our designers to have some visual dashboard to really fast get an overview what the most used (mobile?) screen sizes are 🤷‍♀️), but I fetched some data for tv.nrk.no (and perhaps it includes nrksuper.no as well) that shows how our users map out over different screen sizes ⬇️ I kept it in the mobile range to keep the table from growing too big (and take it with a grain of salt 🤓), but let me know if any of this is helpful 👍 

From 21st of Jan 2026:

Screen width incremented by 10px | Percentage of users
-- | --
0 | 0.01
180 | 0
210 | 0
230 | 0
250 | 0
270 | 0
280 | 0
300 | 0
310 | 0
320 | 0,84
330 | 0,17
340 | 0,03
350 | 0.02
360 | 2,07
370 | 2,29
380 | 2.64
390 | 8.98
400 | 1.16
410 | 04.03
420 | 0.77
430 | 01.03
440 | 0.5
450 | 0.01
460 | 0
480 | 0.01
490 | 0
500 | 0.01
510 | 0
520 | 0

**Comment by ingvilddev** [comment]
> Maybe these breakpoints?
> 
> Size	em	px
> xs	40em	640px
> s	48em	768px
> m	64em	1024px
> l	80em	1280px
> xl	96em	1536px

I am curious how these set breakpoints fit into the idea of [smart layouts](https://www.figma.com/deck/NewdIWNVLbsX8ipYOFZ4p8/Smart-Layouts---CSS-Day-2025?node-id=2330-1964&p=f&viewport=44%2C-78%2C0.03&t=RB3Tz4rgswyOw04s-1&scaling=min-zoom&content-scaling=fixed&page-id=0%3A1) by Ahmad Shadeed, who did a talk showcasing this on [CSS Day](https://cssday.nl/) 2025,  exemplified in the article [Better CSS layouts: Time.com Hero Section](https://ishadeed.com/article/time-layout/) how he would use that way of thinking on actually implement that kind of layout.

The suggested breakpoints are starting with a big number, but from his talks, I've understood it as you should think breakpoints on a component level too, and he starts already at 220px here in the article [The Too Early Breakpoint - why we shouldn't switch to the smallest design too early](https://ishadeed.com/article/too-early-breakpoint/?) (see image ⬇️).

I don't know why the suggested breakpoints are starting from 640px upwards, is it because the thinking is that most users using the public sites (especially if using "Designsystemet" for internal tools) will be on a desktop? 🤔 

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/559d8b32-04c8-45f0-a767-d6a3519edc12" />

I think I want my designers to think in terms of "smart layouts" (so I more easily can write CSS as I understand it is the way to write CSS nowadays with its new capabilities, more "futureproof" perhaps), so if we want to use "Designsystemet", I need it to perhaps reflect that kind of thinking... 🤔 🤓 

I don't know necessarily my intent with this post, but I just understood this issue as a place to share perhaps relevant links... 🤷‍♀️ 🙈

**Comment by mimarz** [comment]
> I don't know why the suggested breakpoints are starting from 640px upwards, is it because the thinking is that most users using the public sites (especially if using "Designsystemet" for internal tools) will be on a desktop? 🤔

To be honest, I have no idea 😂 Its close to two years ago, so wouldn't be surprised if its outdated. Probably from the pitfall of created once and the just copied every use.

This topics open for research, so thanks for sharing your findings, especially your statistics on screen sizes will come in hand. 🤗

**Comment by mimarz** [comment]
We are planning on making an article for this in the not to distant future. Would you be willing to review the article/PR when we get that far? @ingvilddev 

Its always nice to get more perspective on things :)
